plugin/pkg/auth/authenticator/token/oidc/oidc.go

```diff
@@ -155,8 +156,19 @@ func (a *OIDCAuthenticator) AuthenticateToken(value string) (user.Info, bool, er
 		username = fmt.Sprintf("%s#%s", a.clientConfig.ProviderConfig.Issuer, claim)
 	}
 
-	// TODO(yifan): Add UID and Group, also populate the issuer to upper layer.
-	return &user.DefaultInfo{Name: username}, true, nil
+	// TODO(yifan): Add UID, also populate the issuer to upper layer.
+	info := &user.DefaultInfo{Name: username}
+
+	if a.groupsClaim != "" {
+		groups, ok, err := claims.StringsClaim(a.groupsClaim)
```

Comment:
The only error scenario is when the claim is not an array of strings.

```
{}                               // No groups claim, no error.
{"groups": ["group1", "group2"]} // Good response, no error.
{"groups": "group1"}             // Malformed groups claim, this causes an error.
```

Idea being that if authn sees a custom groups claim, it's particular about how that claim is formatted.

We could just log an error and continue though, either way.